.github/workflows/pythonpublish.yml

```diff
@@ -25,15 +23,35 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install setuptools build twine
+          pip install setuptools build
 
       - name: Build
         run: |
           python -m build
 
-      - name: Publish
-        env:
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
-        run: |
-          twine upload dist/*
+      - name: "Upload dists"
+        uses: "actions/upload-artifact@v4"
```

Comment:
Should the actions be pinned to an exact version?

Reply:
Good point, they should all be a little more specific